Concurrency.md

```diff
@@ -0,0 +1,69 @@
+# Concurrency
+
+In general, follow the concurrency section in the [official guidelines](https://docs.swift.org/swift-book/LanguageGuide/Concurrency.html).
+
+## Convention
+
+Use a task to manage synchronous access to a property populated by an an asynchronous function within an Actor. For more in-depth discussions, watch [WWDC21 - Protect mutable state with Swift actors](https://developer.apple.com/wwdc21/10133).
+
+## Rationale
+Even though Actors are helpful for sharing information between concurrent code, asynchronous functions in Actors does not guarantee the state of Actor properties after an asynchronous call.
+
+## Examples
+
+### Bad: when calling the function retrieveURL() across different threads, it's possible that multiple downloads are executed which is inefficient and unnecessary. If the download function always downloads to the same directory, the returned URL might point to an invalid file because another thread has overwritten the url with partially downloaded contents.
+
+``` swift
```

Comment:
Is there a way in this example we could show the sample code accessing this property and printing it so we could show the order and contents of what is printed to clarify the potential badness?

Reply:
Modified the image cache example a little bit to show the potential badness (file contents might be incomplete) in addition to the inefficiency of the "bad" example.